src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionAvoidNoArgumentSuperConstructorCallTest.java

```diff
@@ -62,4 +62,30 @@ public void testOne() throws Exception {
 
         runVerifications(moduleConfig, fileToProcess, expectedViolation, expectedXpathQueries);
     }
+
+    @Test
+    public void testWithComments() throws Exception {
+        final File fileToProcess = new File(getPath(
+            "SuppressionXpathRegressionAvoidNoArgumentSuperConstructorCallWithComments.java"
+        ));
+
+        final DefaultConfiguration moduleConfig =
+                createModuleConfig(CLASS);
+
+        final String[] expectedViolation = {
+            "5:9: " + getCheckMessage(CLASS,
+                AvoidNoArgumentSuperConstructorCallCheck.MSG_CTOR),
+        };
+
+        final List<String> expectedXpathQueries = Collections.singletonList(
```

Comment:
This is the same path as in the first test, thus the same test case. Could you create and test a more complex case like this one:
```
public class A {
  void fun() {
    class B {
      B() {
        super(); // warn
      }
    }
  }
}
```

Reply:
Sure

Reply:
Done, please check